### ISSUE TITLE
Rewrite to Solid.js

### ISSUE THREAD

**Issue by jonathandewitt-dev** [Thunder-Solutions/Thunder-Events#5]
We can keep the native web component idea, but let's not try to reinvent the wheel with our own janky rendering system. We can do better. [Solid does support custom elements](https://github.com/solidjs/solid/tree/main/packages/solid-element#readme), let's see if there's a way to take advantage of that.

**Comment by jonathandewitt-dev** [comment]
Now that we're finished with the [UX mockups](https://www.figma.com/proto/64zuUqJom2YUAgj70YgnsF/thunder-events?node-id=13-415&scaling=min-zoom&page-id=0%3A1&starting-point-node-id=13%3A415) that came from issue #13, we are unblocked on this and free to move forward any time.

**Comment by jonathandewitt-dev** [comment]
Worth making a note here that this rewrite _should not include_ the `<table>` element this time. We can implement the semantics manually without being limited by the structure and style of the native table. See aria roles for [table](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/table_role#description) and [grid](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/grid_role#description).

Using different display types on a table remove the semantics along with the style. The aria role can bring those semantics back. At that point, we might as well use `<div>`, and looking at third party libraries, it seems that's the common trend.

I will _happily_ remove the annoying style and structure limitations in favor of this approach, but we'll have to explore what's actually possible to avoid structure-wise. These roles may still expect a structure like this:

```html
<row>
  <column></column>
  <column></column>
</row>
```

... in which case, some limitations may still exist even despite aria roles and divs. It'll take some figuring..

**Comment by jonathandewitt-dev** [comment]
For now, It may be best just to try addressing the redesign with the current code. The "view by location" screen may not be ideal just yet, but we can get _pretty close_, which is all I can ask for until the work on this issue is complete.

Since this creates a new priority on the task list, I will make a new issue specifically for rearranging/redesigning the UI.

**Comment by jonathandewitt-dev** [comment]
Note: We should absolutely avoid dealing with `href` if we can. Hashchanges are global, and we're working in a confined scope. The global environment is unpredictable and unreliable - there may be watchers attached to `href` or `hash`, especially with client side routers, so let's just totally avoid it.

`<area>` tags don't require the `href` attribute, we can work with it. To be honest, we should have zero `<a>` tags by the time we're done.
